### PR TITLE
Docs: show warning when connecting to incompatible CryoSPARC

### PR DESCRIPTION
**Pull request by nfrasser**
Example output:
```
In [1]: from cryosparc.tools import CryoSPARC

In [2]: cs = CryoSPARC(base_port=62000)
<ipython-input-2-3c3de64e7601>:1: UserWarning: CryoSPARC instance localhost:62000 with version v4.4.0 may not be compatible with current cryosparc-tools version v4.3.1.

To install a compatible version of cryosparc-tools:

    pip install --force cryosparc-tools~=4.4.0

Or, if running a CryoSPARC pre-release or private beta:

    pip install --force https://github.com/cryoem-uoft/cryosparc-tools/archive/refs/heads/develop.zip

  cs = CryoSPARC(base_port=62000)

In [3]:
```

### CODE REVIEW

**Review by Puratinamu — APPROVED** [review]

File: cryosparc/tools.py [hunk 1]
```diff
@@ -143,19 +148,47 @@ def __init__(
...
             self.user_id = self.cli.get_id_by_email_password(email, password)  # type: ignore
+            cs_version: str = self.cli.get_running_version()  # type: ignore
         except Exception as e:
             raise RuntimeError("Could not complete CryoSPARC authentication with given credentials") from e
 
+        if cs_version and VERSION_REGEX.match(cs_version):
+            minor_cs_version = ".".join(cs_version[1:].split(".")[:2])  # e.g., v4.1.0 -> 4.1
+            minor_tools_version = ".".join(__version__.split(".")[:2])  # e.g., 4.1.0 -> 4.1
+            tools_prerelease_url = "https://github.com/cryoem-uoft/cryosparc-tools/archive/refs/heads/develop.zip"
+            if minor_cs_version != minor_tools_version:
```
Comment (Puratinamu) [inline cryosparc/tools.py:181]: should there be a check/message for major version mismatch as well?

Comment (nfrasser) [reply cryosparc/tools.py:181]: It's checking the first two version components, e.g., "vX.Y", not just the "Y". So a major mismatch will register. The variables could have a different to reflect that they are "Up to the minor version component", though I can't think of a good concise name for that 😅

Comment (Puratinamu) [reply cryosparc/tools.py:181]: ah I see, maybe `major_minor_cs_version` could work. Not big deal though

Comment (nfrasser) [reply cryosparc/tools.py]: Done!

**Pull request merged**